tasks/image-quality-predictor/main.py

```diff
@@ -34,74 +34,99 @@ def fetchArtifacts(validationArtifacts: list[int]) -> list[Artifact]:
 
 def main() -> None:
     taskRun = currentTaskRun()
-
     artifacts = fetchArtifacts(taskRun.parameters["validationArtifacts"])
-    epochs: int = taskRun.parameters["epochs"]
-    batchSize: int = taskRun.parameters["batchSize"]
 
     imageSize: int = taskRun.parameters["imageSize"]
     if imageSize < 224:
         raise ValueError("Image size cannot be lower than 224")
 
     dataset = loadDataset(artifacts)
-    trainData, validData = split(taskRun.parameters["validationPct"], dataset)
 
     # Define transformations for your dataset
     transform = transforms.Compose([
         transforms.Resize((imageSize, imageSize)),
         transforms.ToTensor()
     ])
 
-    # Assuming you have your dataset loaded and split into train and test sets
-    trainDataset = ImageQualityDataset(trainData, transform)
-    trainLoader = DataLoader(trainDataset, batchSize, shuffle = True)
+    if not taskRun.parameters["validation"]:
```

Comment:
Extract bodies of `if`/`else` into two functions; `train` and `validate` and call them where appropriate